libmscore/score.cpp

```diff
@@ -3027,6 +3027,11 @@ void Score::selectSimilar(Element* e, bool sameStaff)
                   pattern.subtype = -1; // hack
             else
                   pattern.subtype = e->subtype();
+            pattern.subtypeValid = true;
+            }
+      else if (type == ElementType::TEXT) {
```

Comment:
This is not good, it would remove the ability to select all text at once and only select by subtype. This is already possible with "select > more..."

Reply:
Could you clarify?  The purpose of this PR was to by default only select Text by SubType.  Is that what you say is not good?  Or are you saying that there is an unintentional side effect?

Reply:
Without testing it, it sure *looks* like this would make it impossible to literally select all elements of type TEXT (thus, for example, title, composer, and lyricist all at once).  Are you saying this is actually possible somehow after the change?  if so, what would the new steps be from a user perspective?  Currently, right click one, Select / All Similar Elements does the job, but your PR appears to change this so doing this on title would result in *only* the title being selected, not composer and lyricist as is the case currently.  So an existing capability is lost unless you know of another way to achieve it.

Of course, it is worth questioning whether this particular bit of functionality is *worth* preserving; probably most of the time you'd really only want all Title text selected (eg, in an album).  But worth understanding what is really going on.

Reply:
If you want to select all Titles then you already can, Select > more > Same Subtype.

*The purpose of this PR was to by default only select Text by SubType. Is that what you say is not good?*
Yes, I believe it's not good. It's already possible via Select > More and I believe it's better to have two possibilities: select all text and select by subtype. 

My understanding is that this PR also add the ability to select only measure numbers. That's a valuable feature, which is not available right now in "Select > more" but it doesn't need to modify the behavior of "selectSimilar" and "selectSimilar in range".

Reply:
ok.  Well I think that selecting all of a title text should not select generated elements like tuplet numbers and measure numbers.  I think that because the user never inputted them, and because there is no way to select tuplet numbers.  But I understand that you don't want to change teh select all behavior, and I can accept that.

Maybe instead what could be done would be to recategorize measure numbers and tuplet numbers to something other than "Frame Text" (which is what 3.0 inspector calls them), since to me they clearly don't belong to frame text.  That way selecting all of a frame text like composer/title/lyricist/subtitle would only select all frame text and not measure numbers and tuplet text.

> "My understanding is that this PR also adds the ability to select only measure numbers. That's a valuable feature, which is not available right now in "Select > more" "

Ok, I will spin that part off as a separate PR.  As it is now I already closed this PR, so it is closed for sure.

Reply:
In latest code every text element returns its style id as subtype. This means that measure number and tuplet number now have their own subtype.

Reply:
I tried werner's latest commit https://github.com/musescore/MuseScore/commit/ee0b1b889d3edfcfd76c771a3b70b9a70e95fab6 and it seems to do as he says.  Inspector for a measure number indicates that its substyle is "Measure Number", and I am able to go Select -> More... and only select substyle of measure numbers.  So there is nothing for me to do.